presto-geospatial/src/test/java/com/facebook/presto/plugin/geospatial/aggregation/TestUnionAgg.java

```diff
@@ -0,0 +1,399 @@
+                },
+                {
+                        "disjoint returns multipoint",
+                        "MULTIPOINT ((1 2), (3 4))",
+                        new String[] {"POINT (1 2)", "POINT (3 4)"},
+                },
+        };
+    }
+
+    @DataProvider(name = "linestring")
+    public Object[][] linestring()
+    {
+        return new Object[][] {
+                {
+                        "identity",
+                        "LINESTRING (1 1, 2 2)",
+                        new String[] {"LINESTRING (1 1, 2 2)", "LINESTRING (1 1, 2 2)", "LINESTRING (1 1, 2 2)"},
+                },
+                {
+                        "empty with non-empty",
+                        "LINESTRING (1 1, 2 2)",
+                        new String[] {"LINESTRING EMPTY", "LINESTRING (1 1, 2 2)"},
+                },
+                {
+                        "overlap",
+                        "LINESTRING (1 1, 2 2, 3 3, 4 4)",
+                        new String[] {"LINESTRING (1 1, 2 2, 3 3)", "LINESTRING (2 2, 3 3, 4 4)"},
+                },
+                {
+                        "disjoint returns multistring",
+                        "MULTILINESTRING ((1 1, 2 2, 3 3), (1 2, 2 3, 3 4))",
+                        new String[] {"LINESTRING (1 1, 2 2, 3 3)", "LINESTRING (1 2, 2 3, 3 4)"},
+                },
+                {
+                        "cut through returns multistring",
+                        "MULTILINESTRING ((1 1, 2 2), (3 1, 2 2), (2 2, 3 3), (2 2, 1 3))",
+                        new String[] {"LINESTRING (1 1, 3 3)", "LINESTRING (3 1, 1 3)"},
+                },
+        };
+    }
+
+    @DataProvider(name = "polygon")
+    public Object[][] polygon()
+    {
+        return new Object[][] {
+                {
+                        "identity",
+                        "POLYGON ((2 2, 1 1, 3 1, 2 2))",
+                        new String[] {"POLYGON ((2 2, 1 1, 3 1, 2 2))", "POLYGON ((2 2, 1 1, 3 1, 2 2))", "POLYGON ((2 2, 1 1, 3 1, 2 2))"},
+                },
+                {
+                        "empty with non-empty",
+                        "POLYGON ((2 2, 1 1, 3 1, 2 2))",
+                        new String[] {"POLYGON EMPTY)", "POLYGON ((2 2, 1 1, 3 1, 2 2))"},
+                },
+                {
+                        "three overlapping triangles",
+                        "POLYGON ((1 1, 2 1, 3 1, 4 1, 5 1, 4 2, 3.5 1.5, 3 2, 2.5 1.5, 2 2, 1 1))",
+                        new String[] {"POLYGON ((2 2, 3 1, 1 1, 2 2))", "POLYGON ((3 2, 4 1, 2 1, 3 2))", "POLYGON ((4 2, 5 1, 3 1, 4 2))"},
+                },
+                {
+                        "overlapping geometry collections are simplified",
+                        "POLYGON ((1 1, 2 1, 3 1, 4 1, 5 1, 4 2, 3.5 1.5, 3 2, 2.5 1.5, 2 2, 1 1))",
+                        new String[] {"GEOMETRYCOLLECTION ( POLYGON ((2 2, 3 1, 1 1, 2 2)), POLYGON ((3 2, 4 1, 2 1, 3 2)) )",
```

Comment:
I'd move this test to `geometryCollection()`

Reply:
FYI for some reason Github is showing this as unaddressed, but it's been moved.